### DISCUSSION TITLE
Call Kiali API Status Check with OIDC Login

### DISCUSSION THREAD

**Discussion by cryslam** [kiali/kiali#5615] [Q&A]
Hello! I have enabled OIDC Login and set it up on our openid server...but we're trying to have e2e testing and call kiali api endpoint for status checks to ensure that it's up and running when we do upgrades on our cluster.

Prior we had an internal ldap auth reverse proxy with a programmatic creds in order to make curl calls and check if kiali status is 200 before proceeding further into our service mesh upgrade.

Are there ways to validate kiali status with api when oidc is the choice of auth?

**Comment by jmazzitelli** [comment]
Are you asking to see if a Kiali OIDC login will be successful? Or you just want to see if Kiali status is "up"?

If the latter, you can just go to "/api/status" which returns, in part, a "status" JSON field that looks like this:
```
{
  "status": {
    "Kiali commit hash": "0a808fa70f53867de20e10e052fed182c939c37b",
    "Kiali container version": "v1.59.0",
    "Kiali state": "running",
    "Kiali version": "v1.59.0",
    "Mesh name": "Istio",
    "Mesh version": "1.15.0"
  },
...
}
```

To actually perform an OIDC login is convoluted - you can see how the Kiali molecule test does it (this is what tests that the OpenID auth strategy works inside Kiali): https://github.com/kiali/kiali-operator/blob/master/molecule/openid-test/converge.yml#L53 (you can see performing an OpenID login is not trivial from an API standpoint).

**Reply by cryslam** [reply]
Yes the latter. Previously we were making api call with the programmatic creds with our internal auth method and getting the above response...but now since we switched the curl call to the `api/status` endpoint returns with no authorized since we have enabled the oidc authentication strategy. I'll take a look at the link you've provided, hmm maybe we'll figure out a workaround for this.

thank you!!
